pylops/signalprocessing/Shift.py

```diff
@@ -95,8 +98,24 @@ def Shift(
     else:
         dimsdiag = list(dims)
         dimsdiag[axis] = len(Fop.f)
-    shift = np.exp(-1j * 2 * np.pi * Fop.f * shift)
-    Sop = Diagonal(shift, dims=dimsdiag, axis=axis, dtype=Fop.cdtype)
+
+    shift = _value_or_list_like_to_array(shift)
+
+    if shift.size == 1:
+        shift = np.exp(-1j * 2 * np.pi * Fop.f * shift)
+        Sop = Diagonal(shift, dims=dimsdiag, axis=axis, dtype=Fop.cdtype)
+    else:
+        # add dimensions to shift to match dimensions of model and data
+        if axis < 0:
+            axis = len(dims) + axis
```

Comment:
Consider using `normalize_axis_index` found in `numpy.core.multiarray`

Reply:
Good point, added :)